### ISSUE TITLE
use xcconfig file to facilitate homebrew python build on macOS

### ISSUE THREAD

**Issue by shakfu** [DBraun/DawDreamer#51]
May i suggest to use a [xcconfig](https://nshipster.com/xcconfig/) file to allow for build variations in macOS (such as using Hombrew python). See an example of use [here](https://github.com/shakfu/py-js/blob/master/source/py/targets/shared-ext/py-js.xcconfig) in one my projects.

Otherwise, I am forced to modify the Xcode file directly with this recipe instead of just modifying an `xcconfig` file.

To Compile DawDreamer Xcode using Homebrew Python 3.9.x

1. create JUCER xcode project for macos

2. from the output of `python3-config --cflags` add
   `-I/usr/local/opt/python@3.9/Frameworks/Python.framework/Versions/3.9/include/python3.9`
   to `HEADER_SEARCH_PATHS`

3. from the output of `python3-config --ldflags` add
`-L/usr/local/opt/python@3.9/Frameworks/Python.framework/Versions/3.9/lib/python3.9/config-3.9-darwin`
    to `LIBRARY_SEARCH_PATHS`

4. change -lpython to -lpython3.9 in `OTHER_LDFLAGS` (Other Linker Flags)

**Comment by DBraun** [comment]
Thanks for showing me xcconfig. It's an interesting idea, but I think I want to stay close to the Projucer way for now.

Could you try this? Set the environment variables:
```bash
pythonLocation=/usr/local/opt/python@3.9/Frameworks/Python.framework/Versions/3.9
PYTHONMAJOR=3.9
```

Both of these vars are used in the DawDreamer.jucer file right now, so they need to be set.

Then run from command line:
`xcodebuild -configuration Release -project Builds/MacOSX/DawDreamer.xcodeproj/`
and so on.

If this works for you I'll update the [Developer's Guide](https://github.com/DBraun/DawDreamer/wiki/Developer's-Guide) to explain the case of using Python from homebrew.

**Comment by shakfu** [comment]
Fair enough. Here's a way to do it without environmental variables but with some slight modification to the variables specified in the `.jucer` file:

```
Header Search Paths
    $(PYTHONLOCATION)/include/python$(PYTHONMAJOR);

Extra Library Search Paths
    $(PYTHONLOCATION)/lib
```

Note that I have changed `pythonLocation` to `PYTHONLOCATION` to indicated its is variable and also note that it is now wrapped in  $-brackets. `PYTHONMAJOR` is unchanged.

Then after generating the Xcode project, one can build as follows:

```
xcodebuild -configuration Release -project Builds/MacOSX/DawDreamer.xcodeproj PYTHONMAJOR=3.9 PYTHONLOCATION=/usr/local/opt/python@3.9/Frameworks/Python.framework/Versions/3.9
```

This builds successfully using my homebrew python3.9

S
